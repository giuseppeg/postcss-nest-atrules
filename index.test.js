const postcss = require('postcss')

const plugin = require('./')

function run(input) {
  return postcss([plugin()])
    .process(input)
    .then(result => result.css)
}

it('leaves rules untouched', async () => {
  const css = await run(`
    .a {
      color: red;
    }
  `)

  expect(css).toMatchSnapshot()
})

it('simple', async () => {
  const css = await run(`
    .a { color: red; }
    @media (min-width: 5px) {
      .a { color: hotpink; }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('at-rule alone', async () => {
  const css = await run(`
    @media (min-width: 5px) {
      .a { color: hotpink; }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('nested at-rules', async () => {
  const css = await run(`
    .a { display: block }
    @media (min-width: 50px) {
      @support (display: flex) {
        .a { color: red; }
      }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('at-rules alone', async () => {
  const css = await run(`
    @media (min-width: 50px) {
      @support (display: flex) {
        .a { color: red; }
      }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('at-rules alone (inverse order)', async () => {
  const css = await run(`
    @support (display: flex) {
      @media (min-width: 50px) {
        .a { color: red; }
      }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('multiple rules', async () => {
  const css = await run(`
    .a { color: red; }
    @media (min-width: 5px) {
      .a { color: hotpink; }
      .b { color: green }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('compound selectors', async () => {
  const css = await run(`
    .a, .b { color: red; }
    @media (min-width: 5px) {
      .a { color: hotpink; }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('many occourences', async () => {
  const css = await run(`
    .a { color: red; }
    .a { color: orange; }
    .b { color: blue }

    @media (min-width: 3em) {
      .a {
        color: green
      }

      .b {
        color: yellow;
      }
    }
  `)

  expect(css).toMatchSnapshot()
})

it('big mess', async () => {
  const css = await run(`
    .a { color: red; }
    .a { color: orange; }
    .b { color: blue }

    @media (min-width: 3em) {
      .a {
        color: green
      }

      .b {
        color: yellow;
      }
    }

    .c, .d { color: red }

    @media (min-width: 50px) {
      .c { color: hotpink }
      .d { color: hotpink }
      @support (display: flex) {
      .a { color: red }
      }
      .a { color: red }
    }

    @support (display: flex) {
      @media (min-width: 50px) {
    .a { color: red }
      }
    }
  `)

  expect(css).toMatchSnapshot()
})
