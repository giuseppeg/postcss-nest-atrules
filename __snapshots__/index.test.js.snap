// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`at-rule alone 1`] = `
"
      .a {
    @media (min-width: 5px) { color: hotpink
    } }
  "
`;

exports[`at-rules alone (inverse order) 1`] = `
"
        .a {
    @supports (display: flex) {
      @media (min-width: 50px) { color: red
      }
    } }
  "
`;

exports[`at-rules alone 1`] = `
"
        .a {
    @media (min-width: 50px) {
      @supports (display: flex) { color: red
      }
    } }
  "
`;

exports[`big mess 1`] = `
"
    .a { color: red; }
    .a { color: orange;

    @media (min-width: 3em) {
        color: green
    }

    @media (min-width: 50px) {
      @supports (display: flex) { color: red
      }
    }

    @media (min-width: 50px) { color: red
    }

    @supports (display: flex) {
      @media (min-width: 50px) { color: red
      }
    } }
    .b { color: blue;

    @media (min-width: 3em) {
        color: yellow
    } }

    .c, .d { color: red }

    .c {

    @media (min-width: 50px) { color: hotpink
    } }

    .d {

    @media (min-width: 50px) { color: hotpink
    } }
  "
`;

exports[`compound selectors 1`] = `
"
    .a, .b { color: red; }
    .a {
    @media (min-width: 5px) { color: hotpink
    } }
  "
`;

exports[`does not nest other at rules like @keyframes and @font-face 1`] = `
"
    @keyframes fade {0% { opacity:0 } 100% { opacity:1}}
  "
`;

exports[`leaves rules untouched 1`] = `
"
    .a {
      color: red;
    }
  "
`;

exports[`many occourences 1`] = `
"
    .a { color: red; }
    .a { color: orange;

    @media (min-width: 3em) {
        color: green
    } }
    .b { color: blue;

    @media (min-width: 3em) {
        color: yellow
    } }
  "
`;

exports[`merges with existing ones 1`] = `
"
    .a { color: blue;
    @media (min-width: 3em) {
        display: block
    } }
  "
`;

exports[`multiple rules 1`] = `
"
    .a { color: red;
    @media (min-width: 5px) { color: hotpink
    } }
    .b {
    @media (min-width: 5px) { color: green
    } }
  "
`;

exports[`nested at-rules 1`] = `
"
    .a { display: block;
    @media (min-width: 50px) {
      @supports (display: flex) { color: red
      }
    } }
  "
`;

exports[`simple 1`] = `
"
    .a { color: red;
    @media (min-width: 5px) { color: hotpink
    } }
  "
`;

exports[`without parent 1`] = `
"
      .a {
    @media (min-width: 3em) { display: inline-block
    } }
.b {@media (min-width: 3em) {
        :focus { background-color: yellow; }
    }
      }
  "
`;
